### PR TITLE
MTDSA-13863: Controller improvements

### PR DESCRIPTION
**Pull request by harry-ritchie**

### CODE REVIEW

File: app/api/controllers/BaseController.scala [hunk 1]
```diff
@@ -14,43 +14,20 @@
  * limitations under the License.
  */
 
-package v1.controllers
+/*package api.controllers
```
Comment (tom-lloyd-hmrc) [inline app/api/controllers/BaseController.scala:17]: This file looks to be commented out, can we just remove it?

Comment (harry-ritchie) [reply app/api/controllers/BaseController.scala:17]: Yes we can!

File: app/api/controllers/AuditHandler.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        queryParams = queryParams,
+        requestBody = requestBody,
+        `X-CorrelationId` = ctx.correlationId,
+        auditResponse = auditResponse
+      )
+
+      AuditEvent(auditType, transactionName, detail)
+    }
+
+    auditService.auditEvent(auditEvent)
```
Comment (tom-lloyd-hmrc) [inline app/api/controllers/AuditHandler.scala:75]: This is the auditing step I was a little confused about, as `auditEvent` returns a `Future[AuditResult]`, but we only return a `Unit` so there's no way of checking whether the audit event was actually sent successfully. It's not specific to this PR as these audit results are also ignored in the standard controller pattern, so I've asked in the tech channel whether this is intentional or not.

Comment (tom-lloyd-hmrc) [reply app/api/controllers/AuditHandler.scala:75]: The package internally logs and increments metrics, which is why we don't need to do anything. See [thread](https://hmrcdigital.slack.com/archives/C03DB1GCXB4/p1674644257062989).

File: test/api/controllers/BaseControllerSpec.scala [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package v1.controllers
+/*package api.controllers
```
Comment (tom-lloyd-hmrc) [inline test/api/controllers/BaseControllerSpec.scala:17]: This can be deleted too.

**Review by tom-lloyd-hmrc — APPROVED** [review]

**Pull request merged**